teps/0044-decouple-task-composition-from-scheduling.md

```diff
@@ -0,0 +1,954 @@
+happens, we take that as the cue to schedule those Tasks together.
+
+For example given this Pipeline:
+
+```yaml
+kind: Pipeline
+metadata:
+  name: build-test-deploy
+spec:
+ workspaces:
+  - name: source-code
+  - name: test-results
+ tasks:
+ - name: get-source
+   workspaces:
+   - name: source-code
+     workspace: source-code
+   taskRef:
+     name: git-clone
+   params:
+   - name: url
+      value: $(params.url)
+    - name: revision
+      value: $(params.revision)
+ - name: run-unit-tests
+   runAfter: get-source
+   taskRef:
+     name: just-unit-tests
+   workspaces:
+   - name: source-code
+     workspcae: source-code
+   - name: test-results
+     workspace: test-results
+ - name: upload-results
+   runAfter: run-unit-tests
+   taskRef:
+     name: gcs-upload
+   params:
+   - name: location
+     value: gs://my-test-results-bucket/testrun-$(taskRun.name)
+   workspaces:
+   - name: data
+     workspace: test-results
+```
+
+Running with this PipelineRun would cause `get-source` and `run-unit-tests` to be run in one pod, with `upload-results`
+in another:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: run
+spec:
+  pipelineRef:
+    name: build-test-deply
+  workspaces:
+  - name: source-code
+    emptyDir: {}
+  - name: test-results
+    persistentVolumeClaim:
+      claimName: mypvc
+```
+
+Running with this PipelineRun would cause all of the Tasks to be run in one pod:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: run
+spec:
+  pipelineRef:
+    name: build-test-deply
+  workspaces:
+  - name: source-code
+    emptyDir: {}
+  - name: test-results
+    emptyDir: {}
+```
+
+Running with this PipelineRun would cause all of the Tasks to be run in separate pods:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: run
+spec:
+  pipelineRef:
+    name: build-test-deply
+  workspaces:
+  - name: source-code
+    persistentVolumeClaim:
+      claimName: otherpvc
+  - name: test-results
+    persistentVolumeClaim:
+      claimName: mypvc
+```
+
+Pros:
+* Allows runtime decisions about scheduling without changing the Pod
+
+Cons:
+* If it's important for a Pipeline to be executed in a certain way, that information will have to be encoded somewhere
+  other than the Pipeline 
+
+#### PipelineRun: flag
+
+This is similar to the `emptyDir` based solution but instead of adding extra meaning to `emptyDir` we add a flag to the
+runtime workspace information or to the entire PipelineRun (maybe when this flag is set workspaces do not need to be
+provided.)
+
+#### Controller level
+
+This option is [TEP-0046](https://github.com/tektoncd/community/pull/318). In this option, the Tekton controller can
+be configured to always execute Pipelines inside one pod.
+
+Pros:
+* Making it possible to execute a Pipeline in a pod will also pave the way to be able to support use cases such as
```

Comment:
I think this might be a pro of multiple possible solutions described here - quite a few of them require that a pipeline will need to be fully runnable as a pod.

Reply:
good point, ill add this at the top of the alternatives section as well